network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/WgVpnNetworkStack.kt

```diff
@@ -89,11 +95,13 @@ class WgVpnNetworkStack @Inject constructor(
                 )
             }
 
+            val privateDns = privateDnsProvider.getPrivateDns()
             Result.success(
                 VpnTunnelConfig(
                     mtu = netPDefaultConfigProvider.mtu(),
                     addresses = wgTunnelData?.tunnelAddress ?: emptyMap(),
-                    dns = wgTunnelData!!.allDns(),
+                    // when Android private DNS are set we're force to use them otherwise Android will kill connectivity for that network interface
+                    dns = if (privateDns.isEmpty()) wgTunnelData!!.allDns() else privateDns.toSet(),
```

Comment:
If we want AppTP to also block based on DNS when NetP is enabled, we probably need to make some changes inside our wireguard C/Go module. We can tackle that after we verify that DNS-based blocking works well for AppTP-only users.

Reply:
Yeah, that's correct.

Reply:
Done